app/protocol/flows/blockrelay/handle_relay_invs.go

```diff
@@ -71,7 +72,7 @@ func (flow *handleRelayInvsFlow) start() error {
 		if err != nil {
 			return err
 		}
-		if blockInfo.Exists {
+		if blockInfo.Exists && blockInfo.BlockStatus != externalapi.StatusHeaderOnly {
```

Comment:
Not sure this is correct.
If we receive an inv of a block that is header-only, we are either in IBD (and then we are going to skip this inv in another branch of the code), or it's already pruned.

Why would we want to request it?

Reply:
The top blocks of the DAG can be header only if IBD didn't finish, and then you can get an orphan that will request these blocks.